spec/tasks/gen/task_spec.cr

```diff
@@ -0,0 +1,49 @@
+require "../../spec_helper"
+
+include CleanupHelper
+include GeneratorHelper
+
+private def run_with_args(args : Array(String))
+  generator = Gen::Task.new
+  generator.output = IO::Memory.new
+  generator.print_help_or_call args
+  generator.output
+end
+
+describe Gen::Task do
+  it "generates a task" do
+    with_cleanup do
+      output = run_with_args ["search.reindex"]
+      output.to_s.should contain("Generated ./tasks/search/reindex.cr")
```

Comment:
I notice that we're adding `to_s` calls in a lot of places with this newer injected IO. It may be an easy win to cleanup the interface so that `should_create_files_with_contents` and friends expect a string rather than an IO.